linker/Linker.Steps/MarkStep.cs

```diff
@@ -1701,6 +1785,29 @@ protected virtual void DoAdditionalMethodProcessing (MethodDefinition method)
 		{
 		}
 
+		protected virtual bool ShouldMarkAsInstancePossible (MethodDefinition method)
+		{
+			// We don't need to mark it multiple times
+			if (Annotations.IsInstantiated (method.DeclaringType))
+				return false;
+
+			if (method.IsConstructor && !method.IsStatic)
```

Comment:
If the type ctor it not marked we could convert all instance methods to throw NSE

Reply:
It's on my radar.

I hacked in logging for this scenario in the SweepStep in my `master-base-sweeping` branch where I'm doing base type sweeping (sorting out final details now, it will follow this PR).   Once you have interface + base sweeping the number of methods that fall into this situation is small for `Console.Writeline("Hello world")` program.  Still worth doing, but I wouldn't expect anything significant to come from it.

Here's the list of instance methods, with bodies, on types with no instance ctor that survive sweeping.  And keep in mind we apply some some additional mscorlib.dll preservations.  So this list would be the same or smaller for normal mono + monolinker.
```
T1 System.Tuple`4::get_Item1()
T2 System.Tuple`4::get_Item2()
T3 System.Tuple`4::get_Item3()
T4 System.Tuple`4::get_Item4()
System.Reflection.MethodInfo System.Delegate::get_Method()
System.Object System.Delegate::get_Target()
System.Reflection.MethodInfo System.Delegate::GetMethodImpl()
System.Delegate[] System.Delegate::GetInvocationList()
System.Delegate System.Delegate::CombineImpl(System.Delegate)
System.Delegate System.Delegate::RemoveImpl(System.Delegate)
System.Delegate System.DelegateSerializationHolder/DelegateEntry::DeserializeDelegate(System.Runtime.Serialization.SerializationInfo,System.Int32)
System.MarshalByRefObject System.Runtime.Remoting.Proxies.ProxyAttribute::CreateInstance(System.Type)
System.Runtime.Remoting.Proxies.RealProxy System.Runtime.Remoting.Proxies.ProxyAttribute::CreateProxy(System.Runtime.Remoting.ObjRef,System.Type,System.Object,System.Runtime.Remoting.Contexts.Context)
System.RuntimeType System.Runtime.Remoting.Proxies.TransparentProxy::GetProxyType()
System.Boolean System.Runtime.Remoting.Proxies.TransparentProxy::get_IsContextBoundObject()
System.Runtime.Remoting.Contexts.Context System.Runtime.Remoting.Proxies.TransparentProxy::get_TargetContext()
System.Boolean System.Runtime.Remoting.Proxies.TransparentProxy::InCurrentContext()
System.Object System.Runtime.Remoting.Proxies.TransparentProxy::LoadRemoteFieldNew(System.IntPtr,System.IntPtr)
System.Void System.Runtime.Remoting.Proxies.TransparentProxy::StoreRemoteField(System.IntPtr,System.IntPtr,System.Object)
System.Runtime.InteropServices.MarshalAsAttribute System.Runtime.InteropServices.MarshalAsAttribute::Copy()
System.Void System.Runtime.InteropServices.SafeBuffer::AcquirePointer(System.Byte*&)
System.Void System.Runtime.InteropServices.SafeBuffer::ReleasePointer()
```

Reply:
FYI

I stumbled on a tricky aspect of instance methods that are marked when no ctor is marked.  It's going to be a bit more complicated than I initially realized.  Ideally, we will defer method body processing until an instance ctor is marked.  Otherwise you end up marking things in a body that you are going to stub out with a throw